SchedulerJobs/SchedulerJobs.Services/AnonymiseHearingsConferencesDataService.cs

```diff
@@ -12,52 +15,90 @@ public interface IAnonymiseHearingsConferencesDataService
+            _logger = logger;
         }
+
         public async Task AnonymiseHearingsConferencesDataAsync()
         {
-            await DeleteUserAsync();
-            await _videoApiClient.AnonymiseConferencesAsync();
-            await _bookingsApiClient.AnonymiseHearingsAsync();
-        }
+            var anonymisationData = await _bookingsApiClient.GetAnonymisationDataAsync();
+
+            if (anonymisationData.HearingIds != null && anonymisationData.HearingIds.Any())
+            {
+                _logger.LogInformation("Hearing ids being processed: {hearingids}", anonymisationData.HearingIds);
+                
+                await _videoApiClient.AnonymiseConferenceWithHearingIdsAsync(new AnonymiseConferenceWithHearingIdsRequest
+                    { HearingIds = anonymisationData.HearingIds });
+                
+                await _videoApiClient.AnonymiseQuickLinkParticipantWithHearingIdsAsync(
+                    new AnonymiseQuickLinkParticipantWithHearingIdsRequest { HearingIds = anonymisationData.HearingIds });
+
+                await _bookingsApiClient.AnonymiseParticipantAndCaseByHearingIdAsync("hearingIds",
+                    anonymisationData.HearingIds);
+            }
 
-        private async Task DeleteUserAsync()
-        {
-            // delete users from AAD.
-            // get users that do not have hearings in the future and have had hearing more than 3 months in the past. 
-            // (exclude judges, vhos, test users, performance test users.
-            var usersToDelete = await _bookingsApiClient.GetPersonByClosedHearingsAsync();
 
-            if (usersToDelete != null && usersToDelete.Usernames != null)
+            if (anonymisationData.Usernames == null) return;
+            
+            foreach (var username in anonymisationData.Usernames)
             {
-                foreach (var username in usersToDelete.Usernames)
-                {
+                var userProfile = await _userApiClient.GetUserByAdUserNameAsync(username);
+
+                if (RemoveUserFromAd(userProfile))
                     try
                     {
-                        var user = await _userApiClient.GetUserByAdUserNameAsync(username);
-                        if (user.UserRole == AzureAdUserRoles.VhOfficer || user.UserRole == AzureAdUserRoles.StaffMember || user.IsUserAdmin)
-                            continue;
                         await _userApiClient.DeleteUserAsync(username);
                     }
                     catch (UserApiException exception)
                     {
-                        if (exception.StatusCode != (int)HttpStatusCode.NotFound && exception.StatusCode != (int)HttpStatusCode.Forbidden)
-                        {
-                            throw;
-                        }
+                        _logger.LogError(exception, ProcessingUsernameExceptionMessage,
+                            username);
                     }
+
+                try
+                {
```

Comment:
Can these two try..catch be a single try..catch with a switch on the exception type

Reply:
as I have mentioned this in the earlier review comment, this would mean the whole process will stop if any one of the components throw an error and I thought the idea was we would like it to continue to process other endpoints even if one of them throws an error?